src/common/hooks/useCountdown/index.tsx

```diff
@@ -0,0 +1,32 @@
+import { useState, useEffect, useCallback } from "react";
+
+export const useCountdown = (): {
+  secondsLeft: number | undefined;
+  startCountdown: (seconds: number) => void;
+} => {
+  const [secondsLeft, setSecondsLeft] = useState<number | undefined>(undefined);
+
+  useEffect(() => {
+    if (
+      secondsLeft === undefined ||
+      (secondsLeft !== undefined && secondsLeft <= 0)
```

Comment:
Don’t need !== undefined here again since if it’s undef it will short circuit at first check. Also might be better to just use isNan(secondsleft)?